src/TransCreator/Keyboard.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect } from 'react';
+import { View } from 'react-native';
+import { Button } from 'react-native-elements';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import Calculator, { keyboardLayout } from './calculator';
+import styles from './Style';
+import DateSelector from './DateSelector';
+
+export default function Keyboard({
+  calculable,
+  setTransExpression,
+  setTransAmount,
+  createHandler,
+  transDate,
+  setTransDate,
+}) {
+  const [calculator, setCalculator] = useState(null);
+  useEffect(() => setCalculator(new Calculator()), []);
+
+  const pressHandler = (val) => {
+    if (!calculable) {
+      return;
+    }
+
+    if (val === 'Add') {
+      createHandler();
+    } else if (val === 'C') {
+      calculator.init();
+    } else if (val === 'remove') {
+      calculator.pop();
+    } else {
+      calculator.push(val);
+    }
+    setTransAmount(calculator.getResult());
+    setTransExpression(calculator.getOperation());
+  };
+
+  const numberBtn = (val) => (
+    <Button
+      type="outline"
+      buttonStyle={styles.keyboardBtn}
+      title={val.toString()}
+      titleStyle={{ color: 'rgb(45,45,67)' }}
```

Comment:
This color has been used multiple times, will it better to save it somewhere else and import it to here? Also, it may better to use hex

Reply:
fixed